File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/util/Streams.java
package org.apache.commons.fileupload.util;

import java.io.InputStream;
import java.io.OutputStream;
import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.apache.commons.fileupload.InvalidFileNameException;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public final class Streams {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/util/Streams.py", "Streams"));
  public static final int DEFAULT_BUFFER_SIZE = 8192;

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  public static String checkFileName(String fileName) {

    try {

      if (fileName != null && fileName.indexOf('\u0000') != -1) {
        final StringBuilder sb = new StringBuilder();
        for (int i = 0; i < fileName.length(); i++) {
          char c = fileName.charAt(i);
          switch (c) {
            case 0:
              sb.append("\\0");
              break;
            default:
              sb.append(c);
              break;
          }
        }
        throw new InvalidFileNameException(fileName, "Invalid file name: " + sb);
      }
      return fileName;

    } catch (Throwable ExceptionObjectForCaching) {
      ExceptionHandler.ERR = ExceptionObjectForCaching;
      throw ExceptionObjectForCaching;
    }
  }

  public Streams() {}

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {

    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {

    this.obj.invokeMember("__setattr__", "javaObj", this);

    return idMap;
  }
}

File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/util/Streams.py
from __future__ import annotations
import io
from src.main.org.apache.commons.fileupload.InvalidFileNameException import *
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class Streams(metaclass=StaticFieldRedirector):
    @staticmethod

    def checkFileName(fileName: str) -> str:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        translatedArg0 = JavaHandler.valueToObject(fileName, "String", idMapPyToJ)
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(Streams.javaClz.checkFileName(translatedArg0), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass

            JavaHandler.mapping(translatedArg0, idMapJToPy, fileName)
    def __init__(self) -> None:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        
        idMapJToPy = dict()
        try:
            self.javaObj = Streams.javaClz()
            self.javaObj.setPythonObject(self)
            self.javaObj.jToPy(idMapJToPy)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            
    javaClz = java.type("org.apache.commons.fileupload.util.Streams")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))